packages/flutter_tools/test/integration.shard/transition_test_utils.dart

```diff
@@ -63,7 +66,9 @@ abstract class Transition {
 }
 
 class Barrier extends Transition {
-  const Barrier(this.pattern, {super.handler, super.logging});
+  Barrier(this.pattern, {super.handler, super.logging}) : super(contains: false);
+  Barrier.contains(this.pattern, {super.handler, super.logging}) : super(contains: true);
```

Comment:
A `Pattern` can already express "contains" by using a regex. Why not take the existing pattern and nest it in a regex, e.g. `(.*)($origRegex)(.*)`. (Assuming that substring matching gets compiled efficiently by the regex compiler.)

Reply:
A simple `String.contains(String)` shouldn't turn into a complex `RegExp` match. Also using existing `RegExp` (which uses "contains") for matching a string would require me to escape all regexp-using characters in the string - which is complicated &  error prone (it would make this PR much more complicated) and it would still leave this inconsistency in behavior.

The root cause here is that the *existing code is problematic*: When you give it a `RegExp` it uses "contains" logic when you give it a `String` it uses "equality" logic.

Now by adding a `.contains()` method it actually will perform "contains" operation whether it's given a `RegExp` or a `String` - which seems like a good thing. So the method I add is much cleaner.

(Yes, it still leaves the inconsistency in the old API as it was before - because I don't want to break the API - just extend it with "String contains" functionality. One could clean that existing part up by making it consistently match for equality - also requiring the `RegExp` to match the entire string).

Reply:
> One could clean that existing part up by making it consistently match for equality

Maybe leave a TODO or consider doing a search-replace on all current uses to make the regex ones use the `contains` constructor.

Reply:
I don't want to change all existing usages (otherwise I'd fixed the inconsistency). I added a note (not a TODO because a TODO requires a name and I'm not signing up for it).